util.go

```diff
@@ -9,3 +16,170 @@ func StringPtr(s string) *string {
 func BoolPtr(b bool) *bool {
 	return &b
 }
+
+// containsString returns true if the given list contains the given element.
+func containsString(strings []string, targetString string) bool {
+	for _, aString := range strings {
+		if aString == targetString {
+			return true
+		}
+	}
+	return false
+}
+
+type RateFilterRate struct {
```

Comment:
issue: can we rename this? Why do we repeat `Rate` in the name? We should be more descriptive.

Additionally, this needs to be at the top of the file. Structs come before functions.

Reply:
It's a minified rate that goes in the rate filter. It's a rate-filter rate :)
Any better suggestions?

Reply:
Resolved

Reply:
Your alternative name used is great, thanks